API.md

```diff
@@ -90,3 +90,65 @@ _Note: The `message` field is meant only to describe the error to a user, this t
 
 `{"error":{"id":"no_login","message":"You are not logged in to an account"},"status":"error"}`
 
+### `/resetpass` - Reset a password for a user
+#### Params
+- `account`
+- `email`
+
+#### Responses
+##### Success
+
+`{"status":"ok"}`
+
+#### Errors
+
+`{"error":{"id":"no_account","message":"Unable to find matching account"},"status":"error"}`
+
+`{"error":{"id":"mail_failed","message":"Unable to send reset email"},"status":"error"}`
+
+### `/resetpass/confirm` - Confirm a password reset
+This has two main uses:
+- Simply confirming the reset request and logging in the user, which will then mark the account as unconfirmed until the user changes their password, or
+- Confirming the reset request and also setting the password to `newpass` in one request (requires sending the `newpass` parameter
+
+This allows finer control from the API consumer side as to the separation of authentication and modification of the account
+
+#### Params
+- `account` - Account name to reset the password for
+- `code` - Confirmation code from the user
+- `newpass` _optional_ - If set, sets the new password in this request, otherwise a later request to `/user/set/password` is required.
```

Comment:
newpass should not be optional, as it provides the ability for a user to get locked out.

Reply:
No it doesn't, the user can still log in with the old password or just do a new reset request.

Reply:
Then it leaves an ineffectual API endpoint.

Reply:
How? If you don't set `newpass` it just logs you in to the account without setting the password, returns the session which can be used to change the password with `/user/set/password`

Reply:
Yeah but what happens if the user never actually sets it from then? you have an account without a password. Or the session is lost for $InsertReasonHere? how do you fix it then? another reset? that seems like a whole lot of effort

Reply:
The passsword isn't cleared, it just stays as the old password

Reply:
So yes, ineffectual API endpoint.

Reply:
Resolved by b5186745fa7d0415ff92250d613457e3024bbcc7